pkg/provider/azure/action/linux/imageref.go

```diff
@@ -60,7 +62,24 @@ func getImageRef(osTarget OSType, arch string, version string) (*imageReference,
 			offer:     ir.offer,
 			sku:       fmt.Sprintf(ir.sku, versions[0], versions[1]),
 		}, nil
+	case Fedora:
+		return &imageReference{}, nil
 	}
 	return nil, fmt.Errorf("os type not supported")
 
 }
+
```

Comment:
Is it possible to unify the struct I was using and this other way the ImageID name formatted?

Reply:
we can add another field to the `imageReference` struct:
```go
type imageReference struct {
	publisher string
	offer     string
	sku       string
        // community gallery image ID
        id string
}

```

which would be empty for the existing images, but will have a value for `Fedora` and then while using it in `virtual-machine.go` we could check for `len(imageReferece.id) > 0` and use only the `id` if it exists, otherwise the existing `publisher`, `offer` and `sku` is used for selecting the image

Reply:
Great!!